include/swift/Frontend/FrontendOptions.h

```diff
@@ -116,6 +116,8 @@ class FrontendOptions {
 
     /// Build from a swiftinterface, as close to `import` as possible
     CompileModuleFromInterface,
+    /// Same as CompileModuleFromInterface, but stopping after typechecking
+    TypecheckModuleFromInterface,
```

Comment:
Great addition overall! Why not we just reuse `CompileModuleFromInterface` action instead of defining a new one? We could potentially add another flag to indicate stopping after typechecking for `-compile-module-from-interface`. Maintaining an additional frontend action doesn't seem appealing to me.

Reply:
`-typecheck` and `-compile-module-from-interface` are both compiler modes (`ModeOpt`s), so they're mutually exclusive. Also note that their behavior is pretty different—if you take a look at the FrontendOptions.cpp diff, `ActionType::TypecheckModuleFromInterface`'s behavior tends to match `ActionType::Typecheck`, not `ActionType::CompileModuleFromInterface`.

There's something to be said for the idea of removing `-compile-module-from-interface` and instead having people use `-emit-module` to do that; either there would be a flag to restart the compilation in the module interface builder, or this would just happen automatically when you passed a module interface. This would have a couple of nice benefits, like allowing other flags such as `-dump-ast` or `-emit-sil`. But that would be a much bigger change and I'm not sure it can be fully justified.

(Ultimately, I think the new driver probably ought to treat `.swiftinterface` inputs as implicitly being requests to either compiler or typecheck the interface—not sure which. But that's a whole different story, and I think it's better handled in the new driver instead of the old one.)